File: CORREDOR/Tipos Constructores/JS/Perro.js
const Animal = require("./Animal");

// Constructor CON parámetros 
class Perro extends Animal {
    constructor(nombre, edad, raza, peso) {
        super(nombre, edad); // llama al constructor de la clase abstracta
        this.raza = raza;
        this.peso = peso;
    }

    // Implementación del método de la interfaz simulada
    sonido() {
        console.log("Guau, guau");
    }

    // Implementación del método abstracto
    informacion() {
        console.log(`Nombre: ${this.nombre}`);
        console.log(`Edad: ${this.edad} años`);
        console.log(`Raza: ${this.raza}`);
        console.log(`Peso: ${this.peso} kg`);
    }
}

const perro = new Perro("Firulais", 4, "Labrador", 25);
perro.sonido();     
perro.informacion();

module.exports = Perro;

File: CORREDOR/Tipos Constructores/Java/Perro.java
package Java;

public class Perro extends Animal implements AccionesAnimal {
    String raza;
    double peso;

    // Constructor CON parámetros 
    public Perro(String nombre, int edad, String raza, double peso) {
        super(nombre, edad);// Llama al constructor de Animal
        this.raza = raza;
        this.peso = peso; 
    }

    @Override
    public void sonido(){
        System.out.println("Guau guau...");
    }
    @Override
    void informacion(){
        System.out.println("Nombre: " + nombre);
        System.out.println("Edad: " + edad + " años");
        System.out.println("Raza: " + raza);
        System.out.println("Peso: " + peso + " kg");
    }
}